infra/aws/main.tf

```diff
@@ -52,8 +52,27 @@ module "aws_k8s" {
   vpc_id       = module.aws_base[0].vpc_id
   vpc_subnets  = module.aws_base[0].vpc_subnets
   name_prefix  = "${var.name_prefix}-${var.cluster_id}-${random_string.random_id.result}"
-  cluster_name = var.cluster_name == null ? "eks-${var.aws_k8s_region}-${var.name_prefix}" : var.cluster_name
+  cluster_name = coalesce(var.cluster_name, "eks-${var.aws_k8s_region}-${var.name_prefix}")
   output_path  = var.output_path
   tags         = local.default_tags
   depends_on   = [module.aws_jumpbox[0]]
 }
+
+module "external_dns" {
```

Comment:
This still doesn't feel right. I mean, if it is an _optional addon_, it shouldn't be part fo the main infra module. I expect consumers to be able to do a `make external-dns` the same way they do a `make monitoring` or `make argocd`.
By coupling this here we are just dealing with optional add-ons in two different ways. On the one hand, we have add-ons that appear in the main infra module and you control them with a flag, on the other hand, we have proper add-ons that you explicitly install. This just causes confusion to contributors like me as there is no clear way of adding optional components and makes it hard to evolve the entire thing as now the main modules are coupled to ext-dns no matter what.

I really think this shouldn't be here and external-dns be installed with its dedicate make target like argo or the monitoring stack. This looks more that we're trading developer convenience (at the time of writing the PR) for correctness of the repo structure and that's only going to make things harder to maintain and evolve in the future.

Reply:
I hear you, as I have been also considering below, so in a way looked to DNS as a required infra...

The only reason it is still there, it is desire to get rid of "register_fqdn" step and perhaps have a common approach for dns records management... and eventually have CNAME for UI access... i.e. make it as baseline component, like we have cert-manager.

We need DNS for:

1. TSB MP endpoint (includes XCP)
2. VM onboarding
3. GWs

Even though we can have a separate pointer to external-dns install, I  will have it done.

Reply:
That makes sense